src/metabase/metabot/client.clj

```diff
@@ -60,3 +66,37 @@
             (ex-message e)
             {:exception-data (ex-data e)
              :status-code    400}))))))
+
+(defn- default-embedding-endpoint
+  "OpenAI is the default completion endpoint\""
+  [params options]
+  (openai.api/create-embedding
+   (select-keys params [:model :input])
+   options))
+
+(def ^:dynamic ^{:arglists '([params options])}
+  *create-embedding-endpoint*
+  "Default embeddings endpoint is both dynamic and memoized."
+  (memoize/ttl
+    ^{::memoize/args-fn (fn create-embeddings-memoize-fn [[{:keys [model input]} _options]] [model input])}
+    default-embedding-endpoint
+    ;; 24-hour ttl
+    :ttl/threshold (* 1000 60 60 24)))
+
+(defn create-embedding
+  "Create an embedding vector from the given prompt.
+  This response with the original prompt, the embedding vector, and the token count of the embeddings.
+  The token count can be used to provide best fit queries for prompts requiring large amounts of data."
+  ([model prompt]
+   (log/info "Creating embedding...")
+   (let [{[{:keys [embedding]}]   :data
+          {:keys [prompt_tokens]} :usage} (*create-embedding-endpoint*
+                                            {:model model
+                                             :input prompt}
+                                            {:api-key      (metabot-settings/openai-api-key)
+                                             :organization (metabot-settings/openai-organization)})]
+     {:prompt    prompt
+      :embedding embedding
+      :tokens    prompt_tokens}))
+  ([prompt]
+   (create-embedding "text-embedding-ada-002" prompt)))
```

Comment:
not super necessary but i think we like to extract things like `"text-embedding-ada-002"` into a top level def for clarity.

Reply:
Done